R/databricks_connection.R

```diff
@@ -0,0 +1,64 @@
+databricks_connection <- function(config, extensions) {
+  tryCatch({
+    callSparkR <- get("callJStatic", envir = asNamespace("SparkR"))
```

Comment:
One way we've managed these sorts of "ambient" dependencies (where we want to call something not on CRAN that we *know* is available) is to have the environment primed with a callback. So for example you could set an R option at Notebook startup like this:

```r
options(sparkr.startSparklyr = function() { # do what's needed to start sparklyr and return the port })
```

Then the client code would do this:

```r
startSparklyr <- getOption("sparkr.startSparklyr")
if (!is.null(startSparklyr))
   gatewayPort <- startSparklyr()
```

This is pretty much identical in execution to what we have but it's a bit more decoupled (the implementation of the function called via the `sparkr.startSparklyr` option can be changed at any time). This is also a bit more kosher with CRAN as using asNamespace may work around their automated checks but it's not really in the spirit of not calling packages that aren't on CRAN. Let me know if this makes sense.